File: OCR/ServerSideOCR/ConsoleArchiver/ConsoleArchiver/clsLogging.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace ConsoleArchiver
{
    class clsLogging
    {
        string logDir = System.Configuration.ConfigurationManager.AppSettings["LOGDIR"].ToString ();
        string logFile = "ecm.log." + DateTime.Today.Month.ToString () + "." + DateTime.Today.Day.ToString () + "." + DateTime.Today.Year.ToString () + ".txt";
        
        public clsLogging()
        {
            CkDirExist ();
        }

        public void write(string LogEntry)
        {
            string LogFqn = logDir + @"\" + logFile;
            StreamWriter w = null;
            try
            {
                
                w = File.AppendText ( LogFqn );
                Log ( LogEntry, w );              
            }
            catch (Exception ex)
            {
                Console.WriteLine ( "ERROR: " + ex.Message );
            }
            finally
            {
                w.Close ();
                w.Dispose ();
                GC.Collect ();
                GC.WaitForPendingFinalizers ();
            }           
        }

        void Log ( string logMessage, TextWriter w )
        {            
            w.Write ( "{0} {1}", DateTime.Now.ToLongDateString () , DateTime.Now.ToLongTimeString ());
            w.Write( "  :" );
            w.WriteLine ( "  :{0}", logMessage );
            //w.Write ( "\r\n" );
        }

        void CkDirExist()
        {
            if (!Directory.Exists ( logDir ))
            {
                Directory.CreateDirectory ( logDir );
            }
        }
    }
}

File: ArchiveSVC/clsLogging.vb
Imports System.IO

Public Class clsLogging
    Public Sub WriteToArchiveFileTraceLog(ByVal Msg$, ByVal Zeroize As Boolean)
        Try
            'Dim cPath As String = getTempEnvironDir()
            Dim LogPath As String = System.Configuration.ConfigurationManager.AppSettings("LogPath")
            'Dim TempFolder$ = getEnvVarSpecialFolderApplicationData()
            Dim M$ = Now.Month.ToString.Trim
            Dim D$ = Now.Day.ToString.Trim
            Dim Y$ = Now.Year.ToString.Trim

            Dim SerialNo$ = M + "." + D + "." + Y + "."

            Dim tFQN As String = LogPath + "\ECMLibrary.Archive.FileTrace.Log." + SerialNo$ + "txt"
            If Zeroize Then
                If File.Exists(tFQN) Then
                    File.Delete(tFQN)
                    Return
                End If
            End If
            ' Create an instance of StreamWriter to write text to a file.
            Using sw As StreamWriter = New StreamWriter(tFQN, True)
                ' Add some text to the file.                                    
                sw.WriteLine(Now.ToString + ": " + Msg + vbCrLf)
                sw.Close()
            End Using
        Catch ex As Exception
            Console.WriteLine("clsDma : WriteToSqlLog : 688 : " + ex.Message)
        End Try
    End Sub
End Class
